demos/exyielding/README.md

```diff
@@ -0,0 +1,31 @@
+# Exyielding
+
+In this demo we want to show that a thread can yield the utilization of the CPU directly to another thread.
+
+To illustrate the mechanism in it, we will explain the following:
+* `yield_to` (in `src/user/thread.c`)
+* `yield_next` (in `src/user/thread.c`)
+
+## `yield_to`
+
+By calling this function, the kernel will check if the thread we specified is available to yield to.
+
+If the thread can be yielded, we will set `next_thread` as the address of this thread and trigger `do_scheduler` function.
+
+Since we have decided `next_thread`, the scheduler will do nothing and the kernel will switch the thread.
+
+If the thread we specified is unavailable to yield, `do_scheduler` will not be triggered and we will go back to the point we called `yield_to()`.
+
+## `yield_next`
+
+This function will yield the utilization of the CPU to the next valid thread. The behaviour of this function seems to be similar to the one of `yield()`, but they are a little bit different.
+
+When we calling `yield()`, the scheduler will be triggered and choose the next thread to run. The scheduler may choose the same thread to yield. After switching the context, the thread will go back to the point it yielded before.
+
+In case we are using `yield_next()`, the kernel will walk through `all_threads` array to find the next valid thread to yield. If there is no suitable thread, we will go back to the point we called `yield_next()`. This means that we will not enter `do_scheduler` function and that the context switch will not occur.
+
+## Walkthrough
+
+At the beginning of this demo, we create two named thread  (`first` and `second`). After the thread (`setup`) ends, the thread (`first`) will get the use of the CPU and yield to the thread `second` specified by its `tid`.
```

Comment:
Usually for acronyms I'd say:
"specified by it's thread ID (`tid`)"

Then later you can just use the short version. (and yes I see the irony of ID also being an acronym but I think we can assume people know that one :) )